src/overlays/actors/ovl_Bg_Po_Syokudai/z_bg_po_syokudai.c

```diff
@@ -21,11 +46,128 @@ const ActorInit Bg_Po_Syokudai_InitVars = {
+
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PO_SISTERS, 119.0f, 225.0f, -1566.0f, 0, 0, 0,
+                    this->actor.params);
+        globalCtx->envCtx.unk_BF = 0x4U;
+
+    } else if (!Flags_GetSwitch(globalCtx, 0x1C) && !Flags_GetSwitch(globalCtx, 0x1B)) {
+
+        Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_PO_SISTERS, this->actor.posRot.pos.x,
+                    this->actor.posRot.pos.y + 52.0f, this->actor.posRot.pos.z, 0, 0, 0,
+                    (((u8)this->unk_14C) << 8) + this->actor.params + 0x1000);
+
+    } else if (!Flags_GetSwitch(globalCtx, this->actor.params)) {
+        if (globalCtx->envCtx.unk_BF == 0xFF) {
+            globalCtx->envCtx.unk_BF = 0x4U;
+        }
+    }
+
+    this->unk_14E = (s16)(Math_Rand_ZeroOne() * 20.0f);
+}
+
+void BgPoSyokudai_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    LightContext_RemoveLight(globalCtx, &globalCtx->lightCtx, this->lightNode);
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+
+    if (globalCtx->envCtx.unk_BF != 0xFF) {
+        globalCtx->envCtx.unk_BF = (u8)0xFFU;
+    }
+}
+
+void BgPoSyokudai_Update(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    void* pad;
+
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        func_8002F974(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+    this->unk_14E += 1;
+}
+
+void BgPoSyokudai_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    BgPoSyokudai* this = THIS;
+
+    f32 lightBrightness;
+    u8 red;
+    u8 green;
+    u8 blue;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_bg_po_syokudai.c", 315);
+
+    func_80093D18(globalCtx->state.gfxCtx);
+    gSPMatrix(oGfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_bg_po_syokudai.c", 319),
+              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(oGfxCtx->polyOpa.p++, D_060003A0);
+
+    if (Flags_GetSwitch(globalCtx, this->actor.params)) {
+        Color_RGBA8* primColor = &primColors[this->unk_14C];
+        Color_RGBA8* envColor = &envColors[this->unk_14C];
+
+        lightBrightness = (0.3f * Math_Rand_ZeroOne()) + 0.7f;
+
+        red = ((u32)(f32)(primColor->r * lightBrightness)) & 0xFF;
+        green = ((u32)(f32)(primColor->g * lightBrightness)) & 0xFF;
+        blue = ((u32)((f32)primColor->b * lightBrightness)) & 0xFF;
+
+        Lights_PointSetColorAndRadius(&this->lightInfo, red, green, blue, 0xC8);
+
+        func_80093D84(globalCtx->state.gfxCtx);
+        gSPSegment(oGfxCtx->polyXlu.p++, 0x08,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0U, 0U, 0x20, 0x40, 1, 0,
+                                    ((this->unk_14E) * -20) & 0x1FF, 0x20, 0x80));
```

Comment:
cf `Gfx_TwoTexScroll`'s prototype, most of these arguments make more sense with decimal

can ` & 0x1FF` be replaced with `% 512` ?

this function and its cousins are used to animate textures (scrolling textures) like water for example, here it may be fire?

Reply:
Replacing `& 0x1FF` with `% 512` generates different instructions:

<pre>6dc:    lh      t2,0x14e(t1)                        6dc:    lh      t2,0x14e(t1)                    
6e0:    li      t6,0x80                             6e0:    li      t6,0x80                         
6e4:    li      t5,0x20                             6e4:    li      t5,0x20                         
6e8:    negu    at,t2                               6e8:    negu    at,t2                           
6ec:    sll     t3,at,0x2                           6ec:    sll     t3,at,0x2                       
6f0:    addu    t3,t3,at                            6f0:    addu    t3,t3,at                        
6f4:    sll     t3,t3,0x2                           6f4:    sll     t3,t3,0x2                       
<font color="#729FCF">6f8:    andi    t4,t3,0x1ff</font>                       <font color="#729FCF">| 6f8:    move    v1,v0   </font>                        
<font color="#C4A000">6fc:    sw      </font><font color="#75507B">t4</font><font color="#FCE94F">,0x20(sp)</font>                       <font color="#C4A000">r 6fc:    sw      </font><font color="#75507B">v1</font><font color="#FCE94F">,0x5c(sp)</font>                     
700:    sw      t5,0x24(sp)                         700:    sw      t5,0x24(sp)                     
704:    sw      t6,0x28(sp)                         704:    sw      t6,0x28(sp)                     
708:    move    a1,zero                             708:    move    a1,zero                         
70c:    move    a2,zero                             70c:    move    a2,zero                         
710:    move    a3,zero                             710:    move    a3,zero                         
<font color="#4E9A06">    </font>                                              <font color="#4E9A06">&gt; 714:    bgez    t3,728  </font> <font color="#75507B">~&gt;</font>                     
<font color="#4E9A06">    </font>                                              <font color="#4E9A06">&gt; 718:    andi    t4,t3,0x1ff</font>                     
<font color="#4E9A06">    </font>                                              <font color="#4E9A06">&gt; 71c:    beqz    t4,728  </font> <font color="#75507B">~&gt;</font>                     
<font color="#4E9A06">    </font>                                              <font color="#4E9A06">&gt; 720:    nop     </font>                                
<font color="#4E9A06">    </font>                                              <font color="#4E9A06">&gt; 724:    addiu   t4,t4,-0x200</font>                    
<font color="#C4A000">714:    sw      </font><font color="#06989A">v0</font><font color="#AD7FA8">,0x5c(sp)</font>                       <font color="#C4A000">r 728: </font><font color="#75507B">~&gt;</font> <font color="#C4A000">sw      </font><font color="#06989A">t4</font><font color="#AD7FA8">,0x20(sp)</font></pre>

However, I could create a macro (`MOD512`) if that would make the code easier to read.

Reply:
unfortunate, most `Gfx_TwoTexScroll` calls use `%` (which also means readable code isn't an issue, there are plenty examples to understand that the `&` achieves modulus)